### ISSUE TITLE
e2e is flaky

### ISSUE THREAD

**Issue by clyang82** [openshift-online/maestro#120]
```
• [FAILED] [60.008 seconds]
Resources Resource Delete Option Tests [It] delete the nginx resource from the maestro api [e2e-tests-resources]
/home/runner/work/maestro/maestro/test/e2e/pkg/resources_test.go:109

  [FAILED] Timed out after 60.000s.
  Unexpected error:
      <*errors.StatusError | 0xc000868c80>: 
      deployments.apps "nginx" not found
      {
          ErrStatus: {
              TypeMeta: {Kind: "", APIVersion: ""},
              ListMeta: {
                  SelfLink: "",
                  ResourceVersion: "",
                  Continue: "",
                  RemainingItemCount: nil,
              },
              Status: "Failure",
              Message: "deployments.apps \"nginx\" not found",
              Reason: "NotFound",
              Details: {Name: "nginx", Group: "apps", Kind: "deployments", UID: "", Causes: nil, RetryAfterSeconds: 0},
              Code: 404,
          },
      }
  occurred
  In [It] at: /home/runner/work/maestro/maestro/test/e2e/pkg/resources_test.go:136 @ 06/11/24 08:16:17.353
```

**Comment by clyang82** [comment]
the replicas is 2 which is not desired since the updateStrategy is createOnly. the replicas should remain 1.
```
• [FAILED] [60.009 seconds]
Resources Resource Update Strategy Tests [It] delete the nginx resource [e2e-tests-resources]
/home/runner/work/maestro/maestro/test/e2e/pkg/resources_test.go:200

  [FAILED] Timed out after 60.001s.
  Unexpected error:
      <*errors.errorString | 0xc00090b510>: 
      nginx deployment still exists
      {
          s: "nginx deployment still exists",
      }
  occurred
  In [It] at: /home/runner/work/maestro/maestro/test/e2e/pkg/resources_test.go:217 @ 06/07/24 09:38:02.353
------------------------------
```

**Comment by clyang82** [comment]
it should be fixed in #126.
